code/modules/mob/living/silicon/robot/drone/drone_items.dm

```diff
@@ -141,22 +193,99 @@
 
 				user.visible_message("\red [user] removes the power cell from [A]!", "You remove the power cell.")
 
-//TODO: Matter decompiler.
+
+//NEW MATTER DECOMPILER
 /obj/item/weapon/matter_decompiler
 
 	name = "matter decompiler"
 	desc = "Eating trash, bits of glass, or other debris will replenish your stores."
 	icon = 'icons/obj/device.dmi'
 	icon_state = "decompiler"
-
-	//Metal, glass, wood, plastic.
+	//Temporary storage to assess amount gained per use
+	var/list/store_buffer = list(
+		"metal" = 0,
+		"glass" = 0,
+		"wood" = 0,
+		"plastic" = 0,
+		"plasteel" = 0,
+		"biomass" = 0
+		)
+	//Metal, glass, wood, plastic, plasteel, biomass
 	var/list/stored_comms = list(
 		"metal" = 0,
 		"glass" = 0,
 		"wood" = 0,
-		"plastic" = 0
+		"plastic" = 0,
+		"plasteel" = 0,
+		"biomass" = 0
 		)
 
+/obj/item/weapon/matter_decompiler/verb/check_salvage()
+
+	set name = "Check Salvage Levels"
+	set desc = "Review levels of stored salvage."
+	set category = "Robot Commands"
+	var/S = null
+	var/amount = null
+
+	for(S in src.stored_comms)
+		amount = src.stored_comms[S]
+		src.loc << "\blue The decompiler has [amount] units of [S] stored."
+
+
+/obj/item/weapon/matter_decompiler/verb/recompile_matter()
+
+	set name = "Recompile Matter"
+	set desc = "Recompiles stored matter into sheets."
+	set category = "Robot Commands"
+	var/stored = null
+	var/stackamount = null
+
+	var res_type = input("Select a resource to be recompiled:", "Robot", null, null) in src.stored_comms
```

Comment:
I'm amazed this even managed to pass travis. First, putting nulls there isn't good, those should be filled in with the proper title. Second, it's only even compiling because of relative pathing, which is a big nono, this should be var/res_type. 
Third, minor thing, it none of these references to stored_comms should have src. attached, just reference it by stored_comms.